js/filter.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+window.filter = (function () {
+  var filterContainer = document.querySelector('.map__filters-container');
+  var filterTypeElement = filterContainer.querySelector('#housing-type');
+  var filterPriceElement = filterContainer.querySelector('#housing-price');
+  var filterRoomsElement = filterContainer.querySelector('#housing-rooms');
+  var filterGuestsElement = filterContainer.querySelector('#housing-guests');
+  var filterFeaturesElement = filterContainer.querySelector('#housing-features');
+  var housingFeaturesCheckboxes = document.querySelector('#housing-features').querySelectorAll('.map__checkbox');
+
+  var type;
+  var price;
+  var rooms;
+  var guests;
+  var featuresArray = [];
+  var returnProperty = {};
+
+  function isCorrectType(ad) {
+    if (type === 'any' || ad.offer.type === type) {
+      return ad;
+    }
+    return false;
+  }
```

Comment:
method should return boolean as its name implies. 

you don't need if statements. 

```
function isCorrectType(ad) {
	return type === 'any' || ad.offer.type === type
}
```

Reply:
done